src/components/Table/Table.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { FC, useEffect, useRef, useState } from 'react'
+import styled, { css, DefaultTheme, FlattenInterpolation, ThemeProps } from 'styled-components'
+
+import { deviceBreakPoints } from '../../style/globalStyles'
+
+interface TableProps {
+  main?: boolean
+  hasDetails?: boolean
+  noBorder?: boolean
+  bodyOnly?: boolean
+  scrollable?: boolean
+  isLoading?: boolean
+  minHeight?: number
+}
+
+interface TableLoadingPlaceholderProps {
+  height: number
+}
+
+const Table: FC<TableProps> = ({ children, isLoading, minHeight = 300, ...props }) => {
+  const [height, setHeight] = useState(minHeight)
+  const tableRef = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    const clientHeight = tableRef.current && tableRef.current.clientHeight
+    clientHeight && clientHeight > minHeight && setHeight(clientHeight)
+  }, [minHeight])
+
+  return !isLoading ? (
+    <TableWrapper {...props} ref={tableRef}>
+      <StyledTable {...props}>{children}</StyledTable>
+    </TableWrapper>
+  ) : (
+    <TableLoadingPlaceholder height={height} />
+  )
+}
+
+const TableLoadingPlaceholder = styled.div<TableLoadingPlaceholderProps>`
+  min-height: ${({ height }) => height}px;
+  width: 100%;
+  border-radius: 7px;
+  border: ${({ theme }) => `1px solid ${theme.borderSecondary}`};
+  background-color: ${({ theme }) => theme.bgPrimary};
+  background: linear-gradient(-90deg, rgba(0, 0, 0, 0.05), rgba(255, 255, 255, 0.05), rgba(0, 0, 0, 0.05));
```

Comment:
`background-color` actually acts as a fallback for the few browsers that aren't supporting gradients.

Reply:
Yeah, but `background` overrides `background-color`, whereas `background-image` doesn't. Isn't that the case?

Reply:
See this example: https://codepen.io/eyettea/pen/podZRZg

I assume we want the skeleton loading background with the very low opacity to be displayed ON TOP of the normal background. In that case, setting the shorthand `background` overrides the `background-color` and we do not achieve what we want. 

If we want however to have a transparent background on the element with the skeleton loader so that it gets the background of its parent, them it makes sense (but then why is the fallback background set to anything but `transparent` 🤔 ).